### PR TITLE
Tracing improvements

### LINKED ISSUE #2609: Protocol tracers should use `AnyMessageAndAgency`
Protocol tracers are tracing `AnyMessage` which is missing the state information.  This is confusing for messages which can be send in different states (e.g. blocking / non-blocking mode).

* [x] `chain-sync`
* [x] `block-fetch`
* [x] `tx-submission`
* [x] `local-tx-submission`
* [x] `tip-sample`
* [x] `keep-alive`

### PR DESCRIPTION
**Pull request by coot**
Fixes #2609

### CODE REVIEW

File: ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Type.hs [hunk 1]
```diff
@@ -161,9 +161,10 @@ instance Show (ClientHasAgency (st :: LocalStateQuery block query)) where
   show TokIdle     = "TokIdle"
   show TokAcquired = "TokAcquired"
 
-instance Show (ServerHasAgency (st :: LocalStateQuery block query)) where
-  show TokAcquiring    = "TokAcquiring"
-  show (TokQuerying _) = "TokQuerying"
+instance (forall result. Show (query result))
+    => Show (ServerHasAgency (st :: LocalStateQuery block query)) where
```
[the remark this answers is not in the document]
Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Type.hs:165]: Nice, I'll use it then.

[the remark this answers is not in the document]
Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Type.hs:165]: Done

File: ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Codec.hs [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-{-# LANGUAGE ScopedTypeVariables #-}
-{-# LANGUAGE TypeOperators       #-}
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE NamedFieldPuns        #-}
+{-# LANGUAGE PolyKinds             #-}
+{-# LANGUAGE RankNTypes            #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TypeOperators         #-}
+{-# LANGUAGE QuantifiedConstraints #-}
```
Comment (mrBliss) [inline ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Codec.hs:8]: No longer necessary

**Review by mrBliss — APPROVED** [review]

**Comment by coot** [comment]
bors merge

**Pull request merged**